### PR TITLE
Implement PseudoElementReference.

### PR DESCRIPTION
**Pull request by dstoc**

### CODE REVIEW

**Comment by mithro** [comment]
Need to rebase...

**Comment by dstoc** [comment]
Rebased.

**Comment by mithro** [comment]
LGTM. Assigning to @shans to merge.

File: test/testcases/test-pseudo-element-reference.html [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+var parent = document.querySelector('#parent');
+var child = document.querySelector('#parent');
+document.timeline.play(new Animation(
+    new PseudoElementReference(child, ':before'), {width: ['0px', '100px']}, 1));
+
+timing_test(function() {
+  at(0.5, function() {
+    assert_styles(parent, {height: '20px'});
+  });
+}, 'PseudoElementReference is implmented but not supported.');
```
Comment (shans) [inline test/testcases/test-pseudo-element-reference.html]: speeling mistak

**Pull request merged**